nilearn/glm/second_level/second_level.py

```diff
@@ -843,9 +843,25 @@ def non_parametric_inference(
     # Check design matrix and effect maps agree on number of rows
     _check_effect_maps(effect_maps, design_matrix)
 
+    # Obtain design matrix vars
+    var_names = design_matrix.columns.tolist()
+
     # Obtain tested_var
-    if contrast in design_matrix.columns.tolist():
+    if contrast in var_names:
         tested_var = np.asarray(design_matrix[contrast])
+        var_names.remove(contrast)
+
+    # Obtain confounding vars
+    if len(var_names) == 0:
+        # No other vars in design matrix
+        confounding_vars = None
+    else:
+        # Exclude the intercept (constant)
+        for var_name in var_names:
+            if np.unique(np.asarray(design_matrix[var_name])).size == 1:
```

Comment:
```suggestion
            if np.unique(design_matrix[var_name]).size == 1:
```
I don't think you need that